feat!: Drop webgl1 support (#7453)
maplibre/maplibre-gl-js: CHANGELOG.md

## main
### ✨ Features and improvements
- ⚠️ WebGL (v1) support has been removed; WebGL2 is now required.
  In practical terms, this will not change how you interact with the map.
  This enables performance improvements (e.g. line opacity), Terrain3D enhancements, and several bug fixes.
  WebGL2 support has been widely available for years, and usage of the legacy path had plateaued, so maintaining it no longer justified the added complexity.

>   To ease this breaking change, we have aloso refactored how we handle the case that no webgl is avaliableWebGL (e.g. due to browser restrictions).

Author: ```suggestion
  To ease this breaking change, we have also refactored how we handle the case that no webgl is avaliableWebGL (e.g. due to browser restrictions).
```

- ⚠️ WebGL (v1) support has been removed; WebGL2 is now required.
  In practical terms, this will not change how you interact with the map.
  This enables performance improvements (e.g. line opacity), Terrain3D enhancements, and several bug fixes.
  WebGL2 support has been widely available for years, and usage of the legacy path had plateaued, so maintaining it no longer justified the added complexity.
  To ease this breaking change, we have aloso refactored how we handle the case that no webgl is avaliableWebGL (e.g. due to browser restrictions).
  You can now listen to the webgl error via `.on("error")`, or by overriding `Map._showWebGL2Error`.
  See [caniuse.com/webgl2](https://caniuse.com/webgl2) for ecosystem support and [our RFC for details](https://github.com/maplibre/maplibre-gl-js/discussions/6017). ([#7453](https://github.com/maplibre/maplibre-gl-js/pull/7453)) (by [@CommanderStorm](https://github.com/CommanderStorm))
- Expose `getProjectionData` function in custom layer args objects ([#7471](https://github.com/maplibre/maplibre-gl-js/pull/7471)) (by [@kubapelc](https://github.com/kubapelc))

> - Marked package `sideEffects` as CSS-only in package metadata, which may improve tree-shaking and reduce bundle size in some bundlers ([#7258](https://github.com/maplibre/maplibre-gl-js/pull/7258)) (by [@CommanderStorm](https://github.com/CommanderStorm))

Author: ```suggestion
- ⚠️ WebGL (v1) support has been removed; WebGL2 is now required.
  In practical terms, this will not change how you interact with the map.
  This enables performance improvements (e.g. line opacity), Terrain3D enhancements, and several bug fixes.
  WebGL2 support has been widely available for years, and usage of the legacy path had plateaued, so maintaining it no longer justified the added complexity.
  To ease this breaking change, we have aloso refactored how we handle the case that no webgl is avaliableWebGL (e.g. due to browser restrictions).
  You can now listen to the webgl error via `.on("error")`, or by overriding `Map._showWebGL2Error`.
  See [caniuse.com/webgl2](https://caniuse.com/webgl2) for ecosystem support and [our RFC for details](https://github.com/maplibre/maplibre-gl-js/discussions/6017). ([#7453](https://github.com/maplibre/maplibre-gl-js/pull/7453)) (by [@CommanderStorm](https://github.com/CommanderStorm))
```

## main
### ✨ Features and improvements
- ⚠️ Refactored the `Hash`-based location control (the option that syncs map state to the URL like `#map=5/1/2`) to use `URLSearchParams` internally. This improves extensibility for custom use cases, but may break existing code that relies on the previous implementation. It also changes how certain edge cases are parsed—for example, strings like `#10%2F3.00%2F-1.00` are now accepted, and hashes like `#foo` are normalized to `#foo=`. ([#7073](https://github.com/maplibre/maplibre-gl-js/pull/7073)) (by [@CommanderStorm](https://github.com/CommanderStorm))
- ⚠️ WebGL (v1) support has been removed; WebGL2 is now required.
  In practical terms, this will not change how you interact with the map.
  This enables performance improvements (e.g. line opacity), Terrain3D enhancements, and several bug fixes.
  WebGL2 support has been widely available for years, and usage of the legacy path had plateaued, so maintaining it no longer justified the added complexity.
  To ease this breaking change, we have aloso refactored how we handle the case that no webgl is avaliableWebGL (e.g. due to browser restrictions).

>   The fallback message can be customized via the `.on("error")` handler, or by overriding `Map._showWebGL2Error`.

Author: ```suggestion
  To ease this breaking change, we have aloso refactored how we handle the case that no webgl is avaliableWebGL (e.g. due to browser restrictions).
  You can now listen to the webgl error via `.on("error")`, or by overriding `Map._showWebGL2Error`.
```

>   The fallback message can be customized via our i18n system, the `.on("error")` handler, or by overriding `Map._showWebGL2Error`.

Author: ```suggestion
  The fallback message can be customized via the `.on("error")` handler, or by overriding `Map._showWebGL2Error`.
```
